viadot/sources/vid_club.py

```diff
@@ -0,0 +1,165 @@
+import json
+import urllib
+import os
+
+from typing import Any, Dict, List, Literal
+from datetime import datetime
+
+import pandas as pd
+
+from ..config import local_config
+from ..exceptions import CredentialError, ValidationError
+from ..utils import handle_api_response
+from .base import Source
+
+
+class VidClub(Source):
+    """
+    A class implementing the Vid Club API.
+
+    Documentation for this API is located at: https://evps01.envoo.net/vipapi/
+    There are 4 endpoints where to get the data.
+
+    """
+
+    def __init__(self, *args, credentials: Dict[str, Any] = None, **kwargs):
+        """
+        Create an instance of VidClub.
+
+        Args:
+            credentials (dict, optional): Credentials to Vid Club APIs.
+                Defaults to dictionary.
```

Comment:
Please add `Raises` section in the docstring. In this function, you have `CredentialError`.